awx_collection/plugins/module_utils/tower_api.py

```diff
@@ -154,37 +154,44 @@ def load_config(self, config_path):
 
         # First try to yaml load the content (which will also load json)
         try:
-            config_data = yaml.load(config_string, Loader=yaml.SafeLoader)
-            # If this is an actual ini file, yaml will return the whole thing as a string instead of a dict
-            if type(config_data) is not dict:
-                raise AssertionError("The yaml config file is not properly formatted as a dict.")
+            try_config_parsing = True
+            if HAS_YAML:
+                try:
+                    config_data = yaml.load(config_string, Loader=yaml.SafeLoader)
+                    # If this is an actual ini file, yaml will return the whole thing as a string instead of a dict
+                    if type(config_data) is not dict:
+                        raise AssertionError("The yaml config file is not properly formatted as a dict.")
+                    try_config_parsing = False
 
-        except(AttributeError, yaml.YAMLError, AssertionError):
-            # TowerCLI used to support a config file with a missing [general] section by prepending it if missing
-            if '[general]' not in config_string:
-                config_string = '[general]{0}'.format(config_string)
+                except(AttributeError, yaml.YAMLError, AssertionError):
+                    try_config_parsing = True
 
-            config = ConfigParser()
+            if try_config_parsing:
+                # TowerCLI used to support a config file with a missing [general] section by prepending it if missing
+                if '[general]' not in config_string:
+                    config_string = '[general]{0}'.format(config_string)
```

Comment:
this is going to be a conflict with the fix to add `\n`.

If you don't want to fix conflicts, you can add it here and I'll close the other one.